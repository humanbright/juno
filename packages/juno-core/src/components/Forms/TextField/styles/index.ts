export * from './ClearIconButton';
export * from './OutlineTextFieldStyle';
export * from './TextFieldStyle';
