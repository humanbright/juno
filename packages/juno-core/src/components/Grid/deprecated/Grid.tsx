import React from 'react';

import MuiGrid, { GridProps as MuiGridProps } from '@material-ui/core/Grid';

export type RcGridProps = MuiGridProps;

export const RcGrid: React.SFC<RcGridProps> & {
  dependencies?: any[];
} = React.memo((props) => <MuiGrid {...props} />);

RcGrid.displayName = 'RcGrid';
RcGrid.dependencies = [RcGrid];
