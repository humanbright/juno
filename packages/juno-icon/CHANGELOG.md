
## [1.9.0](https://github.com/ringcentral/juno/compare/juno-icon-v1.8.0...juno-icon-v1.9.0) (2022-06-21)


### Features

* **FIJI-52441:** [Icon] add icon `JumpTo` ([bdc8123](https://github.com/ringcentral/juno/commit/bdc81238038cc031078f222cf3b63821bdb6528d))

## [1.8.0](https://github.com/ringcentral/juno/compare/juno-icon-v1.7.0...juno-icon-v1.8.0) (2022-06-13)


### Features

* **FIJI-51927:** [Icon] add icons: AudioLow, TodayCalendarIco ([ff9a48f](https://github.com/ringcentral/juno/commit/ff9a48fa051790070121bc84e019a4db874efe7d))

## [1.7.0](https://github.com/ringcentral/juno/compare/juno-icon-v1.6.1...juno-icon-v1.7.0) (2022-05-12)


### Features

* **FIJI-50567:** [Icon] add icon HybridContact, update icon AppsCategories, AppsDiscovery, AppsProduct ([c8bd630](https://github.com/ringcentral/juno/commit/c8bd630d0e355ccacd83151f46b756d5e6d6dbc8))

### [1.6.1](https://github.com/ringcentral/juno/compare/juno-icon-v1.6.0...juno-icon-v1.6.1) (2022-05-05)


### Bug Fixes

* **update-icon-220505:** [Icon] fix size of AppsInstalled ([e6d4cc0](https://github.com/ringcentral/juno/commit/e6d4cc0e078ff5e969f430e1ad5430d984e7b9d3))

## [1.6.0](https://github.com/ringcentral/juno/compare/juno-icon-v1.5.0...juno-icon-v1.6.0) (2022-05-05)


### Features

* **update-icon-220505:** [Icon] add icons AppsCategories, AppsDevelopers, AppsDiscovery, AppsInstalled, AppsProduct ([889d824](https://github.com/ringcentral/juno/commit/889d8248a3609f2a74ea07e66081b55709eeabb0))

## [1.5.0](https://github.com/ringcentral/juno/compare/juno-icon-v1.4.0...juno-icon-v1.5.0) (2022-04-28)


### Features

* **FIJI-50006:** [Icon] add icon `shield` ([b255589](https://github.com/ringcentral/juno/commit/b255589e6cbb7a590c38a221432719f557db3535))

### [1.2.1](https://github.com/ringcentral/juno/compare/juno-icon-v1.2.0...juno-icon-v1.2.1) (2022-04-02)


### Bug Fixes

* **update-icon-220402:** [icon] fix icon `record_border` ([9dc7428](https://github.com/ringcentral/juno/commit/9dc74283d7d2b4cd27800ffd83c7251b469890bf))

## [1.1.0](https://github.com/ringcentral/juno/compare/juno-icon-v1.0.3...juno-icon-v1.1.0) (2022-03-17)


### Features

* **FIJI-48126:** [Icon] remove icon `box_border`; update icon `box`; add icon `voicemail_border`, `record_border` ([3152943](https://github.com/ringcentral/juno/commit/3152943032d1c50c6940cd687add13b943ae70c7))

### [1.0.3](https://github.com/ringcentral/juno/compare/juno-icon-v1.0.2...juno-icon-v1.0.3) (2022-03-08)


### Features

* **update-icon-220308:** [Icon] update `Customize` ([5ee9851](https://github.com/ringcentral/juno/commit/5ee9851d0afea20cfe29698b8e282935497f2c1f))

### [1.0.2](https://github.com/ringcentral/juno/compare/juno-icon-v1.0.1...juno-icon-v1.0.2) (2022-03-07)


### Features

* **update-icon-220307:** [Icon] add icon ParkLocation, ParkLocationFull; update icon AvatarRooms ([2409f3f](https://github.com/ringcentral/juno/commit/2409f3f6af389420f7709b9af85ebb6163847099))

### [1.0.1](https://github.com/ringcentral/juno/compare/juno-icon-v1.0.0...juno-icon-v1.0.1) (2022-02-23)


### Features

* **juno-icon:** update-icon ([e741a8d](https://github.com/ringcentral/juno/commit/e741a8db4ebe5ca28e3a30cca93d3bd4c768f1fe))
