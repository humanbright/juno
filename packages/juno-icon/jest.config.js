const baseConfig = require('../../jest-base.config');

module.exports = {
  ...baseConfig,
  moduleNameMapper: {
    '@ringcentral/juno$': '<rootDir>/../juno-core/index.ts',
    '@ringcentral/juno-foundation$':
      '<rootDir>/../juno-core/src/foundation/index.ts',
  },
};
