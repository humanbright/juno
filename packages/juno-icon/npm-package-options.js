module.exports = {
  movePaths: ['./README.md', { from: './assets', to: './' }],
};
