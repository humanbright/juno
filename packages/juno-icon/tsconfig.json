{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "allowJs": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "forceConsistentCasingInFileNames": true,
    "strict": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true
  },
  "files": [],
  "include": []
}
