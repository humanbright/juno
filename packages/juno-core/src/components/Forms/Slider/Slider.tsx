import React, { ComponentProps, forwardRef, useMemo } from 'react';

import MuiSlider from '@material-ui/core/Slider';

import {
  combineClasses,
  RcBaseProps,
  RcPaletteKeys,
  styled,
  useThemeProps,
} from '../../../foundation';
import { SliderStyle } from './styles';
import { RcSliderClasses } from './utils';

type RcSliderProps = {
  /** custom color for Slider */
  color?: RcPaletteKeys;
} & RcBaseProps<ComponentProps<typeof MuiSlider>, 'color'>;

const _RcSlider = forwardRef<any, RcSliderProps>((inProps, ref) => {
  const props = useThemeProps({ props: inProps, name: 'RcSlider' });
  const { classes: classesProp, color, children, ...rest } = props;
  const classes = useMemo(
    () => combineClasses(RcSliderClasses, classesProp),
    [classesProp],
  );

  return (
    <MuiSlider {...rest} ref={ref} classes={classes}>
      {children}
    </MuiSlider>
  );
});

const RcSlider = styled(_RcSlider)`
  ${SliderStyle}
`;

RcSlider.defaultProps = {
  valueLabelDisplay: 'auto',
  color: 'interactive.f01',
};

RcSlider.displayName = 'RcSlider';

export { RcSlider };
export type { RcSliderProps };
