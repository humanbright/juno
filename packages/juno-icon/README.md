# Juno-icon

### Install

```ts
npm i @ringcentral/juno-icon
// or
yarn add @ringcentral/juno-icon
```

## Use Icon

We can use any component with `RcIcon` prop `symbol`, pass component into symbol that will render svg icon with our `RcIcon` style.

```tsx
import { RcIcon } from '@ringcentral/juno';

import { Phone } from '@ringcentral/juno-icon';

const App = (props) => {
  return (
    <RcThemeProvider>
      <RcIcon symbol={Phone} />
    </RcThemeProvider>
  );
};
```
