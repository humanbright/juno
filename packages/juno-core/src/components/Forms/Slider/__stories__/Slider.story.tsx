import React, { ComponentProps, useState } from 'react';

import { SpeakerMute, SpeakerUp, Star } from '@ringcentral/juno-icon';
import {
  notControlInDocTable,
  notShowInDocTable,
  paletteChoice,
  sortInDocTable,
  switchToControlKnobs,
  Title,
} from '@ringcentral/juno-storybook';
import { Meta, Story } from '@storybook/react';

import { RcGrid } from '../../../Grid';
import { RcIcon } from '../../../Icon';
import { RcTextField } from '../../TextField';
import { RcSlider } from '../Slider';

export default {
  title: 'ðŸš€ Cleanup Components/Forms/Slider',
  component: RcSlider,
  argTypes: {
    color: {
      options: paletteChoice,
      control: {
        type: 'select',
      },
    },
    ...sortInDocTable<keyof SliderProps>([
      'defaultValue',
      'valueLabelDisplay',
      'value',
      'onChange',
      'onChangeCommitted',
      'disabled',
      'track',
      'step',
      'marks',
      'orientation',
      'max',
      'min',
      'scale',
    ]),
    ...notControlInDocTable<keyof SliderProps>([]),
    ...notShowInDocTable<keyof SliderProps>([]),
  },
} as Meta;

type SliderProps = ComponentProps<typeof RcSlider>;

export const Slider: Story<SliderProps> = ({ value: valueProp, ...args }) => {
  switchToControlKnobs();
  const [value, setValue] = useState<number | number[]>(valueProp || 0);

  return <RcSlider value={value} onChange={(e, v) => setValue(v)} {...args} />;
};

Slider.args = {};

Slider.argTypes = {
  ...notControlInDocTable<keyof SliderProps>([]),
};

Slider.parameters = {
  tags: [
    // {
    //   name: 'Spec',
    //   href: '',
    // },
    {
      name: 'Mui',
      href: 'https://material-ui.com/components/slider/#slider',
    },
    {
      name: 'Accessibility',
      value: '100%',
    },
  ],
};

export const SliderExamples: Story<SliderProps> = ({
  value: valueProp,
  color,
  ...args
}) => {
  switchToControlKnobs();
  const [value, setValue] = useState<number | number[]>(valueProp || 0);
  const [value2, setValue2] = useState<number | number[]>(valueProp || 0);
  const [value3, setValue3] = useState<number | number[]>(valueProp || 0);

  const handleChange = (
    _e: React.ChangeEvent<{}>,
    v: number | number[],
  ): void => setValue(v);
  const handleChange2 = (
    _e: React.ChangeEvent<{}>,
    v: number | number[],
  ): void => setValue2(v);
  const handleChange3 = (
    _e: React.ChangeEvent<{}>,
    v: number | number[],
  ): void => setValue3(v);

  return (
    <>
      <Title>Slider with icon</Title>
      <RcGrid container spacing={4} alignItems="center">
        <RcGrid item>
          <RcIcon symbol={Star} color={color} size="medium" />
        </RcGrid>
        <RcGrid item xs>
          <RcSlider
            value={value}
            aria-labelledby="input-slider"
            onChange={handleChange}
            color={color}
            {...args}
          />
        </RcGrid>
        <RcGrid item>
          <RcIcon symbol={Star} color={color} size="medium" />
        </RcGrid>
      </RcGrid>
      <br />
      <br />
      <Title>Slider with TextField</Title>
      <RcGrid container spacing={4} alignItems="center">
        <RcGrid item>
          <RcIcon symbol={Star} color={color} size="medium" />
        </RcGrid>
        <RcGrid item xs>
          <RcSlider
            value={value2}
            onChange={handleChange2}
            aria-labelledby="input-slider"
            color={color}
            {...args}
          />
        </RcGrid>
        <RcGrid item>
          <RcTextField
            value={value2}
            style={{ width: '2em' }}
            InputProps={{
              readOnly: true,
            }}
            clearBtn={false}
          />
        </RcGrid>
      </RcGrid>
      <br />
      <br />
      <Title>Slider with Volume</Title>
      <RcGrid container spacing={4} alignItems="center">
        <RcGrid item>
          <RcIcon symbol={SpeakerUp} color={color} size="medium" />
        </RcGrid>
        <RcGrid item xs>
          <RcSlider
            value={value3}
            onChange={handleChange3}
            aria-labelledby="input-slider"
            valueLabelFormat={(x) => `${x}%`}
            color={color}
            {...args}
          />
        </RcGrid>
        <RcGrid item>
          <RcIcon symbol={SpeakerMute} color={color} size="medium" />
        </RcGrid>
      </RcGrid>
    </>
  );
};
