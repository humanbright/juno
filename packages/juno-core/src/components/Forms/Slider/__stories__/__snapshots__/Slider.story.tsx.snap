// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ðŸš€ Cleanup Components/Forms/Slider Slider 1`] = `
.c0 {
  color: #066FAC;
}

.c0.RcSlider-trackInverted .RcSlider-track {
  background-color: rgb(160,200,223);
}

.c0 .RcSlider-thumb.RcSlider-focusVisible {
  box-shadow: 0 0 0 8px rgba(6,111,172,0.16);
}

.c0 .RcSlider-thumb.RcSlider-active {
  box-shadow: 0 0 0 14px rgba(6,111,172,0.16);
}

.c0.RcSlider-disabled {
  color: #B9B9B9;
}

@media (hover:hover) and (pointer:fine) {
  .c0 .RcSlider-thumb:hover {
    box-shadow: 0 0 0 8px rgba(6,111,172,0.16);
  }
}

<span
  className="MuiSlider-root MuiSlider-colorPrimary c0"
  onMouseDown={[Function]}
>
  <span
    className="MuiSlider-rail"
  />
  <span
    className="MuiSlider-track RcSlider-track"
    style={
      Object {
        "left": "0%",
        "width": "0%",
      }
    }
  />
  <input
    type="hidden"
    value="0"
  />
  <span
    aria-orientation="horizontal"
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={0}
    className="MuiSlider-thumb RcSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-217"
    data-index={0}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    role="slider"
    style={
      Object {
        "left": "0%",
      }
    }
    tabIndex={0}
  >
    <span
      className="PrivateValueLabel-offset-219 MuiSlider-valueLabel"
    >
      <span
        className="PrivateValueLabel-circle-220"
      >
        <span
          className="PrivateValueLabel-label-221"
        >
          0
        </span>
      </span>
    </span>
  </span>
</span>
`;

exports[`Storyshots ðŸš€ Cleanup Components/Forms/Slider Slider Examples 1`] = `
Array [
  .c0 {
  font-size: 1.5rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 32px;
  color: #121212;
}

@media (hover:hover) and (pointer:fine) {

}

@media (hover:hover) and (pointer:fine) {

}

@media (hover:hover) and (pointer:fine) {

}

<h5
    className="MuiTypography-root c0 MuiTypography-paragraph MuiTypography-displayBlock"
    data-color="neutral.f06"
    data-variant="headline1"
  >
    Slider with icon
  </h5>,
  .c1 {
  width: 1em;
  height: 1em;
  stroke-width: 0;
  stroke: currentColor;
  fill: currentColor;
  pointer-events: none;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 20px;
}

.c2 {
  color: #066FAC;
}

.c2.RcSlider-trackInverted .RcSlider-track {
  background-color: rgb(160,200,223);
}

.c2 .RcSlider-thumb.RcSlider-focusVisible {
  box-shadow: 0 0 0 8px rgba(6,111,172,0.16);
}

.c2 .RcSlider-thumb.RcSlider-active {
  box-shadow: 0 0 0 14px rgba(6,111,172,0.16);
}

.c2.RcSlider-disabled {
  color: #B9B9B9;
}

@media (hover:hover) and (pointer:fine) {

}

@media (hover:hover) and (pointer:fine) {

}

@media (hover:hover) and (pointer:fine) {
  .c2 .RcSlider-thumb:hover {
    box-shadow: 0 0 0 8px rgba(6,111,172,0.16);
  }
}

<div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <span
        className="c0 star icon"
      >
        <svg
          className="c1"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.885 1.726c.394.195.714.514.908.908l2.97 6.018a1 1 0 0 0 .753.547l6.641.965a2.001 2.001 0 0 1 1.109 3.412L24.46 18.26c-.236.23-.343.561-.288.885l1.134 6.614a1.999 1.999 0 0 1-2.902 2.108l-5.94-3.123a1.002 1.002 0 0 0-.931 0l-5.94 3.123a2 2 0 0 1-2.902-2.109l1.134-6.614a.999.999 0 0 0-.288-.885l-4.806-4.684a2 2 0 0 1 1.109-3.411l6.641-.965a1 1 0 0 0 .753-.547l2.97-6.018a2 2 0 0 1 2.679-.908z"
          />
        </svg>
        
      </span>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
    >
      <span
        className="MuiSlider-root MuiSlider-colorPrimary c2"
        onMouseDown={[Function]}
      >
        <span
          className="MuiSlider-rail"
        />
        <span
          className="MuiSlider-track RcSlider-track"
          style={
            Object {
              "left": "0%",
              "width": "0%",
            }
          }
        />
        <input
          type="hidden"
          value="0"
        />
        <span
          aria-labelledby="input-slider"
          aria-orientation="horizontal"
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={0}
          className="MuiSlider-thumb RcSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-222"
          data-index={0}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          role="slider"
          style={
            Object {
              "left": "0%",
            }
          }
          tabIndex={0}
        >
          <span
            className="PrivateValueLabel-offset-224 MuiSlider-valueLabel"
          >
            <span
              className="PrivateValueLabel-circle-225"
            >
              <span
                className="PrivateValueLabel-label-226"
              >
                0
              </span>
            </span>
          </span>
        </span>
      </span>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <span
        className="c0 star icon"
      >
        <svg
          className="c1"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.885 1.726c.394.195.714.514.908.908l2.97 6.018a1 1 0 0 0 .753.547l6.641.965a2.001 2.001 0 0 1 1.109 3.412L24.46 18.26c-.236.23-.343.561-.288.885l1.134 6.614a1.999 1.999 0 0 1-2.902 2.108l-5.94-3.123a1.002 1.002 0 0 0-.931 0l-5.94 3.123a2 2 0 0 1-2.902-2.109l1.134-6.614a.999.999 0 0 0-.288-.885l-4.806-4.684a2 2 0 0 1 1.109-3.411l6.641-.965a1 1 0 0 0 .753-.547l2.97-6.018a2 2 0 0 1 2.679-.908z"
          />
        </svg>
        
      </span>
    </div>
  </div>,
  @media (hover:hover) and (pointer:fine) {

}

@media (hover:hover) and (pointer:fine) {

}

@media (hover:hover) and (pointer:fine) {

}

<br />,
  @media (hover:hover) and (pointer:fine) {

}

@media (hover:hover) and (pointer:fine) {

}

@media (hover:hover) and (pointer:fine) {

}

<br />,
  .c0 {
  font-size: 1.5rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 32px;
  color: #121212;
}

@media (hover:hover) and (pointer:fine) {

}

@media (hover:hover) and (pointer:fine) {

}

@media (hover:hover) and (pointer:fine) {

}

<h5
    className="MuiTypography-root c0 MuiTypography-paragraph MuiTypography-displayBlock"
    data-color="neutral.f06"
    data-variant="headline1"
  >
    Slider with TextField
  </h5>,
  .c1 {
  width: 1em;
  height: 1em;
  stroke-width: 0;
  stroke: currentColor;
  fill: currentColor;
  pointer-events: none;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 20px;
}

.c3 {
  width: 168px;
}

.c3 .RcTextFieldInputLabel-root {
  font-size: 0.75rem;
  font-weight: 700;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 16px;
  color: #666666;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c3 .RcTextFieldInputLabel-focused {
  color: #066FAC;
}

.c3 .RcTextFieldInput-root {
  color: #121212;
}

.c3 .RcTextFieldInput-input {
  font-size: 1rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 24px;
}

.c3 .RcTextFieldInput-input::-webkit-input-placeholder {
  color: #757575;
  opacity: 1;
}

.c3 .RcTextFieldInput-input::-moz-placeholder {
  color: #757575;
  opacity: 1;
}

.c3 .RcTextFieldInput-input:-ms-input-placeholder {
  color: #757575;
  opacity: 1;
}

.c3 .RcTextFieldInput-input::placeholder {
  color: #757575;
  opacity: 1;
}

.c3 .RcTextFieldInput-underline:before {
  border-bottom-color: #939393;
}

.c3 .RcTextFieldInput-underline:after {
  border-bottom-color: #066FAC;
}

.c3 .RcTextFieldInput-error:after {
  border-bottom-color: #BE3933;
}

.c3 .RcTextFieldInput-disabled {
  color: #B9B9B9;
  -webkit-text-fill-color: #B9B9B9;
}

.c3 .RcTextFieldInput-disabled::-webkit-input-placeholder {
  color: #B9B9B9;
}

.c3 .RcTextFieldInput-disabled::-moz-placeholder {
  color: #B9B9B9;
}

.c3 .RcTextFieldInput-disabled:-ms-input-placeholder {
  color: #B9B9B9;
}

.c3 .RcTextFieldInput-disabled::placeholder {
  color: #B9B9B9;
}

.c3 .RcTextFieldInput-disabled:before {
  border-bottom-style: solid;
  border-bottom-color: #B9B9B9;
}

.c3 .RcTextFieldInput-disabled:after {
  border-bottom-style: none;
}

.c3 .RcTextFieldFormHelperText-root {
  font-size: 0.75rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 16px;
  color: #666666;
}

.c3 .RcTextFieldFormHelperText-error,
.c3 .RcTextFieldInputLabel-error {
  color: #BE3933;
}

.c3 .RcTextFieldFormHelperText-disabled,
.c3 .RcTextFieldInputLabel-disabled {
  color: #B9B9B9;
}

.c3 input[type='search']::-ms-clear,
.c3 input[type='search']::-ms-reveal {
  display: none;
  width: 0;
  height: 0;
}

.c3 input[type='search']::-webkit-search-decoration,
.c3 input[type='search']::-webkit-search-cancel-button,
.c3 input[type='search']::-webkit-search-results-button,
.c3 input[type='search']::-webkit-search-results-decoration {
  display: none;
}

.c2 {
  color: #066FAC;
}

.c2.RcSlider-trackInverted .RcSlider-track {
  background-color: rgb(160,200,223);
}

.c2 .RcSlider-thumb.RcSlider-focusVisible {
  box-shadow: 0 0 0 8px rgba(6,111,172,0.16);
}

.c2 .RcSlider-thumb.RcSlider-active {
  box-shadow: 0 0 0 14px rgba(6,111,172,0.16);
}

.c2.RcSlider-disabled {
  color: #B9B9B9;
}

@media (hover:hover) and (pointer:fine) {

}

@media (hover:hover) and (pointer:fine) {
  .c3 .RcTextFieldInput-underline:hover:not(.RcTextFieldInput-disabled):before {
    border-bottom-color: #121212;
  }
}

@media (hover:hover) and (pointer:fine) {
  .c2 .RcSlider-thumb:hover {
    box-shadow: 0 0 0 8px rgba(6,111,172,0.16);
  }
}

<div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <span
        className="c0 star icon"
      >
        <svg
          className="c1"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.885 1.726c.394.195.714.514.908.908l2.97 6.018a1 1 0 0 0 .753.547l6.641.965a2.001 2.001 0 0 1 1.109 3.412L24.46 18.26c-.236.23-.343.561-.288.885l1.134 6.614a1.999 1.999 0 0 1-2.902 2.108l-5.94-3.123a1.002 1.002 0 0 0-.931 0l-5.94 3.123a2 2 0 0 1-2.902-2.109l1.134-6.614a.999.999 0 0 0-.288-.885l-4.806-4.684a2 2 0 0 1 1.109-3.411l6.641-.965a1 1 0 0 0 .753-.547l2.97-6.018a2 2 0 0 1 2.679-.908z"
          />
        </svg>
        
      </span>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
    >
      <span
        className="MuiSlider-root MuiSlider-colorPrimary c2"
        onMouseDown={[Function]}
      >
        <span
          className="MuiSlider-rail"
        />
        <span
          className="MuiSlider-track RcSlider-track"
          style={
            Object {
              "left": "0%",
              "width": "0%",
            }
          }
        />
        <input
          type="hidden"
          value="0"
        />
        <span
          aria-labelledby="input-slider"
          aria-orientation="horizontal"
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={0}
          className="MuiSlider-thumb RcSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-222"
          data-index={0}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          role="slider"
          style={
            Object {
              "left": "0%",
            }
          }
          tabIndex={0}
        >
          <span
            className="PrivateValueLabel-offset-224 MuiSlider-valueLabel"
          >
            <span
              className="PrivateValueLabel-circle-225"
            >
              <span
                className="PrivateValueLabel-label-226"
              >
                0
              </span>
            </span>
          </span>
        </span>
      </span>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <div
        className="MuiFormControl-root MuiTextField-root RcTextField-root c3"
        style={
          Object {
            "width": "2em",
          }
        }
      >
        <div
          className="MuiInputBase-root MuiInput-root RcTextFieldInput-root MuiInput-underline RcTextFieldInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input RcTextFieldInput-input"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={true}
            required={false}
            type="text"
            value={0}
          />
        </div>
      </div>
    </div>
  </div>,
  @media (hover:hover) and (pointer:fine) {

}

@media (hover:hover) and (pointer:fine) {

}

@media (hover:hover) and (pointer:fine) {

}

<br />,
  @media (hover:hover) and (pointer:fine) {

}

@media (hover:hover) and (pointer:fine) {

}

@media (hover:hover) and (pointer:fine) {

}

<br />,
  .c0 {
  font-size: 1.5rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 32px;
  color: #121212;
}

@media (hover:hover) and (pointer:fine) {

}

@media (hover:hover) and (pointer:fine) {

}

@media (hover:hover) and (pointer:fine) {

}

<h5
    className="MuiTypography-root c0 MuiTypography-paragraph MuiTypography-displayBlock"
    data-color="neutral.f06"
    data-variant="headline1"
  >
    Slider with Volume
  </h5>,
  .c1 {
  width: 1em;
  height: 1em;
  stroke-width: 0;
  stroke: currentColor;
  fill: currentColor;
  pointer-events: none;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 20px;
}

.c2 {
  color: #066FAC;
}

.c2.RcSlider-trackInverted .RcSlider-track {
  background-color: rgb(160,200,223);
}

.c2 .RcSlider-thumb.RcSlider-focusVisible {
  box-shadow: 0 0 0 8px rgba(6,111,172,0.16);
}

.c2 .RcSlider-thumb.RcSlider-active {
  box-shadow: 0 0 0 14px rgba(6,111,172,0.16);
}

.c2.RcSlider-disabled {
  color: #B9B9B9;
}

@media (hover:hover) and (pointer:fine) {

}

@media (hover:hover) and (pointer:fine) {

}

@media (hover:hover) and (pointer:fine) {
  .c2 .RcSlider-thumb:hover {
    box-shadow: 0 0 0 8px rgba(6,111,172,0.16);
  }
}

<div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <span
        className="c0 speaker-up icon"
      >
        <svg
          className="c1"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.414 3.414A2 2 0 0 1 20 4.828v22.343a2 2 0 0 1-3.414 1.414L10 21.999H4a2 2 0 0 1-2-2v-8a2 2 0 0 1 2-2h6l6.586-6.586a2 2 0 0 1 2.828 0zM18 4.828l-6.586 6.586A2 2 0 0 1 10 12H5a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h5a2 2 0 0 1 1.414.586L18 27.172V4.829zm8.003 2.809a1 1 0 0 1 1.395.232C29.081 10.224 30 13.044 30 16s-.919 5.776-2.602 8.131a1 1 0 0 1-1.627-1.163C27.213 20.95 28 18.535 28 16s-.787-4.95-2.229-6.968a1 1 0 0 1 .232-1.395zm-3.521 3.568a1 1 0 0 1 1.362.38A8.963 8.963 0 0 1 25 16.001a8.953 8.953 0 0 1-1.171 4.441 1 1 0 1 1-1.738-.988A6.965 6.965 0 0 0 23 16.001a6.962 6.962 0 0 0-.898-3.433 1 1 0 0 1 .38-1.362z"
          />
        </svg>
        
      </span>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
    >
      <span
        className="MuiSlider-root MuiSlider-colorPrimary c2"
        onMouseDown={[Function]}
      >
        <span
          className="MuiSlider-rail"
        />
        <span
          className="MuiSlider-track RcSlider-track"
          style={
            Object {
              "left": "0%",
              "width": "0%",
            }
          }
        />
        <input
          type="hidden"
          value="0"
        />
        <span
          aria-labelledby="input-slider"
          aria-orientation="horizontal"
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={0}
          className="MuiSlider-thumb RcSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-222"
          data-index={0}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          role="slider"
          style={
            Object {
              "left": "0%",
            }
          }
          tabIndex={0}
        >
          <span
            className="PrivateValueLabel-offset-224 MuiSlider-valueLabel"
          >
            <span
              className="PrivateValueLabel-circle-225"
            >
              <span
                className="PrivateValueLabel-label-226"
              >
                0%
              </span>
            </span>
          </span>
        </span>
      </span>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <span
        className="c0 speaker-mute icon"
      >
        <svg
          className="c1"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.414 3.414A2 2 0 0 1 20 4.828v22.343a2 2 0 0 1-3.414 1.414L10 21.999H4a2 2 0 0 1-2-2v-8a2 2 0 0 1 2-2h6l6.586-6.586a2 2 0 0 1 2.828 0zM18 4.828l-6.586 6.586A2 2 0 0 1 10 12H5a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h5a2 2 0 0 1 1.414.586L18 27.172V4.829zm5.172 6.929 2.83 2.828 2.827-2.828a.999.999 0 1 1 1.414 1.414l-2.827 2.828 2.827 2.828a.999.999 0 1 1-1.414 1.414l-2.827-2.828-2.83 2.828a.999.999 0 1 1-1.414-1.414l2.83-2.828-2.83-2.828a.999.999 0 1 1 1.414-1.414z"
          />
        </svg>
        
      </span>
    </div>
  </div>,
]
`;
